lib/vty.h

```diff
@@ -196,6 +211,34 @@ static inline void vty_push_context(struct vty *vty, int node, uint64_t id)
 	struct structname *ptr = VTY_GET_CONTEXT(structname);                  \
 	VTY_CHECK_CONTEXT(ptr);
 
+/* XPath */
+#define VTY_PUSH_XPATH(nodeval, value)                                         \
+	do {                                                                   \
+		if (vty->xpath_index >= VTY_MAXDEPTH) {                        \
+			vty_out(vty, "%% Reached maximum CLI depth (%u)\n",    \
+				VTY_MAXDEPTH);                                 \
+			return CMD_WARNING;                                    \
+		}                                                              \
+		vty->node = nodeval;                                           \
+		strlcpy(vty->xpath[vty->xpath_index], value,                   \
+			sizeof(vty->xpath[0]));                                \
+		vty->xpath_index++;                                            \
+	} while (0)
+
+#define VTY_GET_XPATH vty->xpath[vty->xpath_index - 1]
```

Comment:
Perhaps this should be `VTY_CURR_XPATH` or similar, I would expect `GET` to be part of a function-like macro.